src/org/apache/commons/httpclient/HttpMethodBase.java

```diff
@@ -1368,6 +1372,84 @@ protected void addCookieRequestHeader(HttpState state, HttpConnection conn)
     }
 
     /**
+     * Put all the cookies in a single header line. 
+     * 
+     * Merge the cookies already present in the request 
+     * with the cookies coming from the state. 
+     * 
+     * @param state the {@link HttpState state} information associated with this method
+     * @param conn the {@link HttpConnection connection} used to execute
+     *        this HTTP method
+     * @param cookies associated with the {@link HttpState state}
+     * 
+     */
+	private void putAllCookiesInASingleHeader(HttpState state, HttpConnection conn,
+			Cookie[] cookies) {
+		
+		LOG.trace("enter putAllCookiesInASingleHeader(HttpState state, HttpConnection conn, Cookie[] cookies)" );
+		
+		String host = getHost(state, conn);
+		CookieSpec matcher = getCookieSpec(state);
+		//use a map to make sure we only have one cookie per name
+		HashMap<String, Cookie> mergedCookies = new HashMap<String, Cookie>();		
+		Header[] cookieheaders = getRequestHeaderGroup().getHeaders("Cookie");
+		for (int i = 0; i < cookieheaders.length; i++) {
+			Header cookieheader = cookieheaders[i];
+			HeaderElement[] headerElements = parseCookieHeaders(cookieheader);
+			for (int j = 0; j < headerElements.length; j++) {
+				Cookie cookie = new Cookie(host,
+						headerElements[j].getName(),
+						headerElements[j].getValue());
+				mergedCookies.put(cookie.getName(),cookie);
+			}
+			// clean the header
+			getRequestHeaderGroup().removeHeader(cookieheader);
+		}
+		//the cookies coming from the state override 
+		//the cookies coming from the request
+		for (Cookie cookie : cookies) {
+			mergedCookies.put(cookie.getName(),cookie);
+		}
+		cookies = mergedCookies.values().toArray(new Cookie[mergedCookies.size()]);
+		String s = matcher.formatCookies(cookies);
+		getRequestHeaderGroup()
+				.addHeader(new Header("Cookie", s, true));
+	}
+	/**
+	 * Parse a cookie header in a list of headerElement.
+	 * 
+	 * Unfortunatly the method Header.getElements() is not working properly.
+	 * On a line like Cookie: has-js=1; drupal-toolbar=off it only return the 
+	 * first element has-js=1. So we rework this one. 
+	 * 
+	 * This is a "won't fix" known issue as the project ended.
+	 * https://issues.apache.org/jira/browse/HTTPCLIENT-1356
+	 *  
+	 * @param cookieheader
+	 * @return a list of name/value pair
```

Comment:
fixed
`@return a list of HeaderElement corresponding to name of the cookie and its value`